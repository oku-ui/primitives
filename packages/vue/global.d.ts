/// <reference types="vite/client" />
import 'vitest'
import type { AxeMatchers } from 'vitest-axe/matchers'

declare module 'vitest' {
  export interface AsymmetricMatchersContaining extends AxeMatchers {}
  export interface Assertion extends AxeMatchers {}
}
