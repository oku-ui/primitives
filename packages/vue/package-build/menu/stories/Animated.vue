<script setup lang="ts">
import { reactive, ref } from 'vue'
import { OkuMenuCheckboxItem, OkuMenuItemIndicator, OkuMenuRadioGroup, OkuMenuRadioItem } from '@oku-ui/menu'
import MenuWithAnchor from './MenuWithAnchor.vue'
import TickIcon from './TickIcon.vue'

const files = ['README.md', 'index.js', 'page.css']
const file = ref(files[1])
const open = ref(true)
const checkboxItems = reactive([
  { label: 'Bold', state: false },
  { label: 'Italic', state: true },
  { label: 'Underline', state: false },
  { label: 'Strikethrough', state: false, disabled: true },
])
</script>

<template>
  <label>
    <input type="checkbox" :checked="open" @change="(event) => open = (event.target as HTMLInputElement).checked">{{ ' ' }}
    open
  </label>
  <br>
  <br>
  <MenuWithAnchor class="menu-animated-content menu-content" :open="open">
    <OkuMenuCheckboxItem
      v-for="({ label, state, disabled }, index) in checkboxItems"
      :key="label"
      class="menu-item"
      :checked="state"
      :disabled="disabled"
      @checked-change="checkboxItems[index].state = !state"
    >
      {{ label }}
      <OkuMenuItemIndicator class="menu-animated-item-indicator">
        <TickIcon />
      </OkuMenuItemIndicator>
    </OkuMenuCheckboxItem>
    <OkuMenuRadioGroup :value="file" @value-change="(value: string) => file = value">
      <OkuMenuRadioItem v-for="_file in files" :key="_file" class="menu-item" :value="_file">
        {{ _file }}
        <OkuMenuItemIndicator class="menu-animated-item-indicator">
          <TickIcon />
        </OkuMenuItemIndicator>
      </OkuMenuRadioItem>
    </OkuMenuRadioGroup>
  </MenuWithAnchor>
</template>
