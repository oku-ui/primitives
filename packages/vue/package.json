{
  "name": "vue-packages",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@rollup/plugin-alias": "^5.1.0"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "^5.0.3",
    "@vueuse/core": "^10.7.2",
    "filter-anything": "^3.0.7",
    "typescript": "^5.3.3",
    "unplugin-vue-components": "^0.26.0",
    "vite-plugin-pages": "^0.32.0",
    "vue-router": "^4.2.5",
    "vue-tsc": "^2.0.0"
  }
}
