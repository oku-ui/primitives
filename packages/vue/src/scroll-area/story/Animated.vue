<script setup lang="ts">
import ScrollAreaStory from './ScrollAreaStory.vue'
import Copy from './Copy.vue'
</script>

<template>
  <ScrollAreaStory animated :style="{ width: '800px', height: '800px' }">
    <Copy v-for="(_, index) in 30" :key="index" />
  </ScrollAreaStory>
</template>
