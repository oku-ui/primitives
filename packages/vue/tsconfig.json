{
  "compilerOptions": {
    "target": "ESNext",
    "jsx": "preserve",
    "jsxImportSource": "vue",
    "lib": [
      "ESNext",
      "DOM",
      "DOM.Iterable"
    ],
    "useDefineForClassFields": true,
    "baseUrl": ".",
    "module": "ESNext",
    "moduleResolution": "bundler",
    "paths": {
      "@/*": [
        "src/*"
      ],
      "@oku-ui/accordion": [
        "src/accordion"
      ],
      "@oku-ui/alert-dialog": [
        "src/alert-dialog"
      ],
      "@oku-ui/arrow": [
        "src/arrow"
      ],
      "@oku-ui/aspect-ratio": [
        "src/aspect-ratio"
      ],
      "@oku-ui/avatar": [
        "src/avatar"
      ],
      "@oku-ui/collection": [
        "src/collection"
      ],
      "@oku-ui/collapsible": [
        "src/collapsible"
      ],
      "@oku-ui/checkbox": [
        "src/checkbox"
      ],
      "@oku-ui/dialog": [
        "src/dialog"
      ],
      "@oku-ui/direction": [
        "src/direction"
      ],
      "@oku-ui/dismissable-layer": [
        "src/dismissable-layer"
      ],
      "@oku-ui/focus-guards": [
        "src/focus-guards"
      ],
      "@oku-ui/focus-scope": [
        "src/focus-scope"
      ],
      "@oku-ui/hover-card": [
        "src/hover-card"
      ],
      "@oku-ui/label": [
        "src/label"
      ],
      "@oku-ui/menu": [
        "src/menu"
      ],
      "@oku-ui/popover": [
        "src/popover"
      ],
      "@oku-ui/popper": [
        "src/popper"
      ],
      "@oku-ui/portal": [
        "src/portal"
      ],
      "@oku-ui/primitive": [
        "src/primitive"
      ],
      "@oku-ui/progress": [
        "src/progress"
      ],
      "@oku-ui/provide": [
        "src/provide"
      ],
      "@oku-ui/radio-group": [
        "src/radio-group"
      ],
      "@oku-ui/roving-focus": [
        "src/roving-focus"
      ],
      "@oku-ui/scroll-area": [
        "src/scroll-area"
      ],
      "@oku-ui/separator": [
        "src/separator"
      ],
      "@oku-ui/slider": [
        "src/slider"
      ],
      "@oku-ui/slot": [
        "src/slot"
      ],
      "@oku-ui/switch": [
        "src/switch"
      ],
      "@oku-ui/tabs": [
        "src/tabs"
      ],
      "@oku-ui/toggle": [
        "src/toggle"
      ],
      "@oku-ui/toggle-group": [
        "src/toggle-group"
      ],
      "@oku-ui/toast": [
        "src/toast"
      ],
      "@oku-ui/toolbar": [
        "src/toolbar"
      ],
      "@oku-ui/tooltip": [
        "src/tooltip"
      ],
      "@oku-ui/use-composable": [
        "src/use-composable"
      ],
      "@oku-ui/utils": [
        "src/utils"
      ],
      "@oku-ui/visually-hidden": [
        "src/visually-hidden"
      ],
      "@oku-ui/presence": [
        "src/presence"
      ]
    },
    "resolveJsonModule": true,
    "allowImportingTsExtensions": true,
    "strict": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noEmit": true,
    "isolatedModules": true,
    "skipLibCheck": true
  },
  "references": [
    {
      "path": "./tsconfig.node.json"
    }
  ],
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "*.d.ts"
  ]
}
