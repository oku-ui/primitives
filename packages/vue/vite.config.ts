import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import Pages from 'vite-plugin-pages'

const resolve = (val: string) => new URL(val, import.meta.url).pathname

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    Pages(),
  ],
  server: {
    fs: {
      // Allow serving files from two level up to the project root
      allow: ['..'],
    },
  },
  build: {
    watch: {
      include: ['../components/**'],
    },
  },
  resolve: {
    alias: [
      // Components
      {
        find: '@oku-ui/scroll-area',
        replacement: resolve('./src/scroll-area'),
      },
      {
        find: '@oku-ui/checkbox',
        replacement: resolve('./src/checkbox'),
      },
      {
        find: '@oku-ui/alert-dialog',
        replacement: resolve('./src/alert-dialog'),
      },
      {
        find: '@oku-ui/slider',
        replacement: resolve('./src/slider'),
      },
      {
        find: '@oku-ui/label',
        replacement: resolve('./src/label'),
      },
      {
        find: '@oku-ui/separator',
        replacement: resolve('./src/separator'),
      },
      {
        find: '@oku-ui/popover',
        replacement: resolve('./src/popover'),
      },
      {
        find: '@oku-ui/toolbar',
        replacement: resolve('./src/toolbar'),
      },
      {
        find: '@oku-ui/toast',
        replacement: resolve('./src/toast'),
      },
      {
        find: '@oku-ui/toggle',
        replacement: resolve('./src/toggle'),
      },
      {
        find: '@oku-ui/toggle-group',
        replacement: resolve('./src/toggle-group'),
      },
      {
        find: '@oku-ui/switch',
        replacement: resolve('./src/switch'),
      },
      {
        find: '@oku-ui/radio-group',
        replacement: resolve('./src/radio-group'),
      },
      {
        find: '@oku-ui/avatar',
        replacement: resolve('./src/avatar'),
      },
      {
        find: '@oku-ui/hover-card',
        replacement: resolve('./src/hover-card'),
      },
      {
        find: '@oku-ui/collapsible',
        replacement: resolve('./src/collapsible'),
      },
      {
        find: '@oku-ui/progress',
        replacement: resolve('./src/progress'),
      },
      {
        find: '@oku-ui/tooltip',
        replacement: resolve('./src/tooltip'),
      },
      {
        find: '@oku-ui/aspect-ratio',
        replacement: resolve('./src/aspect-ratio'),
      },
      {
        find: '@oku-ui/tabs',
        replacement: resolve('./src/tabs'),
      },
      {
        find: '@oku-ui/dialog',
        replacement: resolve('./src/dialog'),
      },
      {
        find: '@oku-ui/primitives',
        replacement: resolve('./src/primitives'),
      },
      {
        find: '@oku-ui/accordion',
        replacement: resolve('./src/accordion'),
      },

      // Core
      {
        find: '@oku-ui/menu',
        replacement: resolve('./src/menu'),
      },
      {
        find: '@oku-ui/dismissable-layer',
        replacement: resolve('./src/dismissable-layer'),
      },
      {
        find: '@oku-ui/roving-focus',
        replacement: resolve('./src/roving-focus'),
      },
      {
        find: '@oku-ui/focus-scope',
        replacement: resolve('./src/focus-scope'),
      },
      {
        find: '@oku-ui/presence',
        replacement: resolve('./src/presence'),
      },
      {
        find: '@oku-ui/direction',
        replacement: resolve('./src/direction'),
      },
      {
        find: '@oku-ui/arrow',
        replacement: resolve('./src/arrow'),
      },
      {
        find: '@oku-ui/visually-hidden',
        replacement: resolve('./src/visually-hidden'),
      },
      {
        find: '@oku-ui/popper',
        replacement: resolve('./src/popper'),
      },
      {
        find: '@oku-ui/collection',
        replacement: resolve('./src/collection'),
      },
      {
        find: '@oku-ui/slot',
        replacement: resolve('./src/slot'),
      },
      {
        find: '@oku-ui/portal',
        replacement: resolve('./src/portal'),
      },
      {
        find: '@oku-ui/focus-guards',
        replacement: resolve('./src/focus-guards'),
      },
      {
        find: '@oku-ui/use-composable',
        replacement: resolve('./src/use-composable'),
      },
      {
        find: '@oku-ui/provide',
        replacement: resolve('./src/provide'),
      },
      {
        find: '@oku-ui/primitive',
        replacement: resolve('./src/primitive'),
      },
      {
        find: '@oku-ui/utils',
        replacement: resolve('./src/utils'),
      },
    ],
  },
})
