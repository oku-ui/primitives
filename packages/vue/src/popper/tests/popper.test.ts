import { afterEach, beforeEach, describe, expect, it, vitest } from 'vitest'
import type { VueWrapper } from '@vue/test-utils'
import { enableAutoUnmount, mount } from '@vue/test-utils'
import { axe } from 'vitest-axe'
import { nextTick } from 'vue'
import {
  OkuPopper,
  OkuPopperAnchor,
  OkuPopperArrow,
  OkuPopperContent,
} from '../'

import StyledVue from '../stories/Styled.vue'
import Animated from '../stories/Animated.vue'
import OneScroll from '../stories/OneScroll.vue'
import Transition from '../stories/Transition.vue'
import WithCustomArrow from '../stories/WithCustomArrow.vue'
import WithPortal from '../stories/WithPortal.vue'

enableAutoUnmount(afterEach)

describe('okuPopper', () => {
  it('okuPopper renders correctly', () => {
    const wrapper = mount(OkuPopper)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('okuPopperArrow renders correctly', () => {
    const spy = vitest.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuPopperArrow)
    expect(() => wrapper()).toThrowError('`OkuPopperArrow` must be used within `OkuPopperContent`')
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuPopperContent)" not found.')
  })

  it('okuPopperAnchor renders correctly', () => {
    const spy = vitest.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuPopperAnchor)
    expect(() => wrapper()).toThrowError('`OkuPopperAnchor` must be used within `OkuPopper`')
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuPopper)" not found.')
  })

  it('okuPopperContent renders correctly', () => {
    const spy = vitest.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => mount(OkuPopperContent)
    expect(() => wrapper()).toThrowError('`OkuPopperContent` must be used within `OkuPopper`')
    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuPopper)" not found.')
  })

  describe('styledVue', () => {
    let wrapper: VueWrapper<InstanceType<typeof StyledVue>>

    beforeEach(async () => {
      wrapper = mount(StyledVue, {
        attachTo: document.body,
      })

      await wrapper.find('[class="popper-anchor"]').trigger('click')
      await nextTick()
    })

    /**
     * @vitest-environment jsdom
     */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(document.body).toMatchSnapshot()
    })

    it('should be able to close', async () => {
      const button = wrapper.find('button')
      await button.trigger('click')
      await nextTick()
      expect(document.body).toMatchSnapshot()
    })
  })

  describe('animated', () => {
    let wrapper: VueWrapper<InstanceType<typeof Animated>>

    beforeEach(async () => {
      wrapper = mount(Animated, {
        props: {
          container: document.body,
        },
        attachTo: document.body,
      })

      await wrapper.find('[class="popper-anchor"]').trigger('click')
      await nextTick()
    })

    /**
     * @vitest-environment jsdom
     */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(document.body).toMatchSnapshot()
    })

    it('document.body has correct html', () => {
      expect(document.body).toMatchSnapshot()
    })

    it('should be able to close', async () => {
      const button = document.querySelector('button')
      button?.dispatchEvent(new Event('click'))
      await nextTick()
      expect(document.body).toMatchSnapshot()
    })
  })

  describe('oneScroll', () => {
    let wrapper: VueWrapper<InstanceType<typeof OneScroll>>

    beforeEach(async () => {
      wrapper = mount(OneScroll, {
        attachTo: document.body,
      })
      await wrapper.find('[class="popper-anchor small"]').trigger('click')
    })

    /**
     * @vitest-environment jsdom
     */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(document.body.innerHTML).toMatchSnapshot()
    })

    it('should be able to close', async () => {
      const button = document.querySelector('button')
      button?.dispatchEvent(new Event('click'))
      await nextTick()
      expect(document.body.innerHTML).toMatchSnapshot()
    })
  })

  describe('transition', () => {
    let wrapper: VueWrapper<InstanceType<typeof Transition>>

    beforeEach(async () => {
      wrapper = mount(Transition, {
        attachTo: document.body,
      })
      await wrapper.find('[class="popper-anchor"]').trigger('click')

      await new Promise(resolve => setTimeout(resolve, 300))
      await nextTick()
    })

    /**
     * @vitest-environment jsdom
     */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(document.body).toMatchSnapshot()
    })

    it('renders body innerHTML correctly', () => {
      expect(document.body.innerHTML).toMatchSnapshot()
    })

    it('should be able to close', async () => {
      const button = document.querySelector('button')
      button?.click()
      await nextTick()
      // TODO: fix this dont see close button
      expect(document.body.innerHTML).toMatchSnapshot()
    })
  })

  describe('withCustomArrow', () => {
    let wrapper: VueWrapper<InstanceType<typeof WithCustomArrow>>

    beforeEach(async () => {
      wrapper = mount(WithCustomArrow, {
        attachTo: document.body,
      })
      await wrapper.find('[class="popper-anchor"]').trigger('click')
    })

    /**
     * @vitest-environment jsdom
     */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(document.body).toMatchSnapshot()
    })
  })

  describe('withPortal', () => {
    let wrapper: VueWrapper<InstanceType<typeof WithPortal>>

    beforeEach(async () => {
      wrapper = mount(WithPortal, {
        attachTo: document.body,
      })

      await wrapper.find('[class="popper-anchor"]').trigger('click')
    })

    /**
     * @vitest-environment jsdom
     */
    it('axe accessibility tests', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('renders correctly', () => {
      expect(document.body).toMatchSnapshot()
    })
  })
})
