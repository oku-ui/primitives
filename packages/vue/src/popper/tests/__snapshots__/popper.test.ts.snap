// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuPopper > animated > document.body has correct html 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div
      container="[object HTMLBodyElement]"
      style="display: flex; align-items: center; justify-content: center; height: 200vh;"
    >
      
      
      
      <div
        class="popper-anchor"
      >
         open 
      </div>
      
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      
    </div>
  </div>
  <div
    data-oku-popper-content-wrapper=""
    style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;"
  >
    <div
      class="popper-animatedContent"
      data-align="center"
      data-side="bottom"
      style="animation: none;"
    >
      <button>
         close 
      </button>
      <span
        style="position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;"
      >
        <svg
          class="popper-arrow"
          height="10"
          offset="25"
          preserveAspectRatio="none"
          style="display: block;"
          viewBox="0 0 30 10"
          width="20"
        >
          <polygon
            points="0,0 30,0 15,10"
          />
        </svg>
      </span>
    </div>
  </div>
  
</body>
`;

exports[`okuPopper > animated > renders correctly 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div
      container="[object HTMLBodyElement]"
      style="display: flex; align-items: center; justify-content: center; height: 200vh;"
    >
      
      
      
      <div
        class="popper-anchor"
      >
         open 
      </div>
      
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      
    </div>
  </div>
  <div
    data-oku-popper-content-wrapper=""
    style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;"
  >
    <div
      class="popper-animatedContent"
      data-align="center"
      data-side="bottom"
      style="animation: none;"
    >
      <button>
         close 
      </button>
      <span
        style="position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;"
      >
        <svg
          class="popper-arrow"
          height="10"
          offset="25"
          preserveAspectRatio="none"
          style="display: block;"
          viewBox="0 0 30 10"
          width="20"
        >
          <polygon
            points="0,0 30,0 15,10"
          />
        </svg>
      </span>
    </div>
  </div>
  
</body>
`;

exports[`okuPopper > animated > should be able to close 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div
      container="[object HTMLBodyElement]"
      style="display: flex; align-items: center; justify-content: center; height: 200vh;"
    >
      
      
      
      <div
        class="popper-anchor"
      >
         open 
      </div>
      
      <!--v-if-->
      
      
    </div>
  </div>
</body>
`;

exports[`okuPopper > okuPopper renders correctly 1`] = `""`;

exports[`okuPopper > oneScroll > renders correctly 1`] = `"<div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""><div style="display: flex; gap: 100px;"><div><h1>In scrolling container</h1><div style="width: 400px; height: 600px; overflow: auto; border: 1px solid black;"><div style="height: 2000px;"><div class="popper-anchor small"> 9 </div><div data-oku-popper-content-wrapper="" style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;"><div data-side="bottom" data-align="center" class="popper-content small" style="animation: none;"><span style="position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;"><svg class="popper-arrow" style="display: block;" width="10" height="5" viewBox="0 0 30 10" preserveAspectRatio="none"><polygon points="0,0 30,0 15,10"></polygon></svg></span> 9 </div></div></div></div></div></div></div>"`;

exports[`okuPopper > oneScroll > should be able to close 1`] = `"<div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""><div style="display: flex; gap: 100px;"><div><h1>In scrolling container</h1><div style="width: 400px; height: 600px; overflow: auto; border: 1px solid black;"><div style="height: 2000px;"><div class="popper-anchor small"> 9 </div><div data-oku-popper-content-wrapper="" style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;"><div data-side="bottom" data-align="center" class="popper-content small" style="animation: none;"><span style="position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;"><svg class="popper-arrow" style="display: block;" width="10" height="5" viewBox="0 0 30 10" preserveAspectRatio="none"><polygon points="0,0 30,0 15,10"></polygon></svg></span> 9 </div></div></div></div></div></div></div>"`;

exports[`okuPopper > styledVue > renders correctly 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div
      style="display: flex; align-items: center; justify-content: center; height: 200vh;"
    >
      
      
      
      <div
        class="popper-anchor"
      >
         open 
      </div>
      
      <div
        data-oku-popper-content-wrapper=""
        style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;"
      >
        <div
          class="popper-content"
          data-align="center"
          data-side="bottom"
          style="animation: none;"
        >
          <button>
             close 
          </button>
          <span
            style="position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;"
          >
            <svg
              class="popper-arrow"
              height="10"
              preserveAspectRatio="none"
              style="display: block;"
              viewBox="0 0 30 10"
              width="20"
            >
              <polygon
                points="0,0 30,0 15,10"
              />
            </svg>
          </span>
        </div>
      </div>
      
      
    </div>
  </div>
</body>
`;

exports[`okuPopper > styledVue > should be able to close 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div
      style="display: flex; align-items: center; justify-content: center; height: 200vh;"
    >
      
      
      
      <div
        class="popper-anchor"
      >
         open 
      </div>
      
      <!--v-if-->
      
      
    </div>
  </div>
</body>
`;

exports[`okuPopper > transition > renders body innerHTML correctly 1`] = `"<div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""><div class="popper-anchor"> Open </div><!--teleport start--><!--teleport end--></div><transition-stub data-v-1f07280c="" name="fade" appear="false" persisted="true" css="true"><div data-v-1f07280c="" data-oku-popper-content-wrapper="" style="position: fixed; left: 0px; top: 0px; transform: translate(5px, 0px); min-width: max-content; --oku-popper-transform-origin: 0px 0px; --oku-popper-available-width: -5px; --oku-popper-available-height: 0px; --oku-popper-anchor-width: 0px; --oku-popper-anchor-height: 0px;"><div data-side="right" data-align="start" class="popper-animatedContent" style=""><button data-v-1f07280c=""> close </button><span data-v-1f07280c="" style="position: absolute; left: 0px; transform-origin: 0 0; transform: translateY(50%) rotate(90deg) translateX(-50%);"><svg class="popper-arrow" offset="25" style="display: block;" width="20" height="10" viewBox="0 0 30 10" preserveAspectRatio="none"><polygon points="0,0 30,0 15,10"></polygon></svg></span></div></div></transition-stub>"`;

exports[`okuPopper > transition > renders correctly 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    
    
    <div
      class="popper-anchor"
    >
       Open 
    </div>
    
    <!--teleport start-->
    <!--teleport end-->
    
  </div>
  <transition-stub
    appear="false"
    css="true"
    data-v-1f07280c=""
    name="fade"
    persisted="true"
  >
    <div
      data-oku-popper-content-wrapper=""
      data-v-1f07280c=""
      style="position: fixed; left: 0px; top: 0px; transform: translate(5px, 0px); min-width: max-content; --oku-popper-transform-origin: 0px 0px; --oku-popper-available-width: -5px; --oku-popper-available-height: 0px; --oku-popper-anchor-width: 0px; --oku-popper-anchor-height: 0px;"
    >
      <div
        class="popper-animatedContent"
        data-align="start"
        data-side="right"
        style=""
      >
        <button
          data-v-1f07280c=""
        >
           close 
        </button>
        <span
          data-v-1f07280c=""
          style="position: absolute; left: 0px; transform-origin: 0 0; transform: translateY(50%) rotate(90deg) translateX(-50%);"
        >
          <svg
            class="popper-arrow"
            height="10"
            offset="25"
            preserveAspectRatio="none"
            style="display: block;"
            viewBox="0 0 30 10"
            width="20"
          >
            <polygon
              points="0,0 30,0 15,10"
            />
          </svg>
        </span>
      </div>
    </div>
  </transition-stub>
  
</body>
`;

exports[`okuPopper > transition > should be able to close 1`] = `"<div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""></div><div data-v-app=""><div class="popper-anchor"> Open </div><!--teleport start--><!--teleport end--></div><transition-stub data-v-1f07280c="" name="fade" appear="false" persisted="true" css="true"><div data-v-1f07280c="" data-oku-popper-content-wrapper="" style="position: fixed; left: 0px; top: 0px; transform: translate(5px, 0px); min-width: max-content; --oku-popper-transform-origin: 0px 0px; --oku-popper-available-width: -5px; --oku-popper-available-height: 0px; --oku-popper-anchor-width: 0px; --oku-popper-anchor-height: 0px; display: none;"><div data-side="right" data-align="start" class="popper-animatedContent" style=""><button data-v-1f07280c=""> close </button><span data-v-1f07280c="" style="position: absolute; left: 0px; transform-origin: 0 0; transform: translateY(50%) rotate(90deg) translateX(-50%);"><svg class="popper-arrow" offset="25" style="display: block;" width="20" height="10" viewBox="0 0 30 10" preserveAspectRatio="none"><polygon points="0,0 30,0 15,10"></polygon></svg></span></div></div></transition-stub>"`;

exports[`okuPopper > withCustomArrow > renders correctly 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div
      style="display: flex; align-items: center; justify-content: center; height: 200vh;"
    >
      
      
      
      <div
        class="popper-anchor"
      >
         open 
      </div>
      
      <div
        data-oku-popper-content-wrapper=""
        style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;"
      >
        <div
          class="popper-content"
          data-align="center"
          data-side="right"
          style="animation: none;"
        >
          <button>
             close 
          </button>
          <span
            style="position: absolute; left: 0px; transform-origin: 0 0; transform: translateY(50%) rotate(90deg) translateX(-50%); visibility: hidden;"
          >
            
            <div
              height="10"
              style="width: 20px; height: 10px; border-bottom-left-radius: 10px; border-bottom-right-radius: 10px; background-color: tomato;"
              width="20"
            />
            
          </span>
        </div>
      </div>
      
      
    </div>
  </div>
</body>
`;

exports[`okuPopper > withPortal > renders correctly 1`] = `
<body>
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  />
  <div
    data-v-app=""
  >
    <div
      style="display: flex; align-items: center; justify-content: center; height: 200vh;"
    >
      
      
      
      <div
        class="popper-anchor"
      >
         open 
      </div>
      
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      
    </div>
  </div>
  <div
    data-oku-popper-content-wrapper=""
    style="position: fixed; left: 0px; top: 0px; transform: translate(0, -200%); min-width: max-content; --oku-popper-transform-origin:  ;"
  >
    <div
      class="popper-content"
      data-align="center"
      data-side="bottom"
      style="animation: none;"
    >
      <button>
         close 
      </button>
      <span
        style="position: absolute; top: 0px; transform-origin: center 0; transform: rotate(180deg); visibility: hidden;"
      >
        <svg
          class="popper-arrow"
          height="10"
          preserveAspectRatio="none"
          style="display: block;"
          viewBox="0 0 30 10"
          width="20"
        >
          <polygon
            points="0,0 30,0 15,10"
          />
        </svg>
      </span>
    </div>
  </div>
  
</body>
`;
